pandas/core/window.py

```diff
@@ -0,0 +1,1339 @@
+        return self._apply_window(mean=False, **kwargs)
+
+    @Substitution(name='rolling')
+    @Appender(_doc_template)
+    def mean(self, **kwargs):
+        return self._apply_window(mean=True, **kwargs)
+
+class _Rolling(_Window):
+
+    @property
+    def _constructor(self):
+        return Rolling
+
+    def _apply(self, func, window=None, center=None, check_minp=None, how=None, **kwargs):
+        """
+        Rolling statistical measure using supplied function. Designed to be
+        used with passed-in Cython array-based functions.
+
+        Parameters
+        ----------
+        func : string/callable to apply
+        window : int/array, default to _get_window()
+        center : boolean, default to self.center
+        check_minp : function, default to _use_window
+        how : string, default to None
+            how to resample
+
+        Returns
+        -------
+        y : type of input
+        """
+        if center is None:
+            center = self.center
+        if window is None:
+            window = self._get_window()
+
+        if check_minp is None:
+            check_minp = _use_window
+
+        blocks, obj = self._create_blocks(how=how)
+        results = []
+        for b in blocks:
+            try:
+                values = self._prep_values(b.values)
+            except TypeError:
+                results.append(b.values.copy())
+                continue
+
+            if values.size == 0:
+                results.append(values.copy())
+                continue
+
+            # if we have a string function name, wrap it
+            if isinstance(func, compat.string_types):
+                if not hasattr(algos, func):
+                    raise ValueError("we do not support this function algos.{0}".format(func))
+
+                cfunc = getattr(algos, func)
+                def func(arg, window, min_periods=None):
+                    minp = check_minp(min_periods, window)
+                    return cfunc(arg, window, minp, **kwargs)
+
+            # calculation function
+            if center:
+                offset = _offset(window, center)
+                additional_nans = np.array([np.NaN] * offset)
+                def calc(x):
+                    return func(np.concatenate((x, additional_nans)),
+                                window, min_periods=self.min_periods)
+            else:
+                def calc(x):
+                    return func(x,window, min_periods=self.min_periods)
+
+            if values.ndim > 1:
+                result = np.apply_along_axis(calc, self.axis, values)
+            else:
+                result = calc(values)
+
+            if center:
+                result = self._center_window(result, window)
+
+            results.append(result)
+
+        return self._wrap_results(results, blocks, obj)
+
+class _Rolling_and_Expanding(_Rolling):
+
+    _shared_docs['count'] = """%(name)s count of number of non-NaN observations inside provided window."""
```

Comment:
It would be nice if all these docstrings would start with a capital letter, but the problem is the same name is used in the See also part, where a capital letter is not OK of course